### PR TITLE
fix numpy dependenc and cleanup recipe.

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
@conda-forge/matplotlib this is ready for review when anybody has time.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,11 +26,11 @@ requirements:
   host:
     - python
     - setuptools
-    - numpy >=1.10
+    - numpy 1.11
```
Comment (ocefpaf) [inline recipe/meta.yaml]: :+1

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,11 +26,11 @@ requirements:
...
+    - cycler 0.10
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Does this one have to be that strict? Can do `>=0.10`?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -51,63 +51,28 @@ requirements:
...
+    - matplotlib.backends._backend_agg
+    - matplotlib.ft2font
+    - matplotlib.ttconv
+    - matplotlib.backends._tkagg  # [not win]
+    - mpl_toolkits
+    - pylab
+
 outputs:
   - name: matplotlib-base
-    script: install.sh  # [not win]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Thanks! I the outputs are still the same, right?

File: recipe/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipe/meta.yaml]: Let's use `1.11.*`.

File: recipe/meta.yaml (hunk 3 above)
Comment (hmaarrfk) [reply recipe/meta.yaml]: Yeah, basically if you name an output the same as the master package, I think it takes in all the rest of the fields.

File: recipe/meta.yaml (hunk 2 above)
Comment (hmaarrfk) [reply recipe/meta.yaml]: I think so. I also think many of the host dependencies are not necessary. matplotlib isn't very good at seperating "build time" and "run time" dependencies.

File: recipe/yum_requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+mesa-libGL
```
Comment (jakirkham) [inline recipe/yum_requirements.txt]: IIUC the new compilers ignore this. Should we add a CDT instead?

Comment (hmaarrfk) [reply recipe/yum_requirements.txt]: Feel free to try. I couldn't get the CDT to work without this.

Comment (hmaarrfk) [reply recipe/yum_requirements.txt]: incidentally, when I added this line, the CDT was no longer necessary

Comment (ocefpaf) [reply recipe/yum_requirements.txt]: Looks like that broke the Linux builds :-/

Comment (hmaarrfk) [reply recipe/yum_requirements.txt]: yeah, i don't know what changed in the anvil image, many GL things seem to be breaking now. I think this really just needs the yum dependency.

Comment (ocefpaf) [reply recipe/yum_requirements.txt]: There is some discussion on gitter about it. I'll check it out tomorrow.
Meanwhile I'm OK just re-adding the yum_requires.

**Comment by ocefpaf** [comment]
Thanks @hmaarrfk! Let's get this merged and we can sort the azure failure out and the `ctd` stuff later.

**Pull request merged**